subsys/bluetooth/host/hci_core.c

```diff
@@ -6007,6 +6026,37 @@ int bt_le_adv_stop(void)
 	return 0;
 }
 
+void bt_le_adv_resume(void)
+{
+	struct bt_conn *adv_conn = NULL;
+	int err;
+
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
```

Comment:
Wont the linker strip it away from the final image if there are no callers of it?

Reply:
If CONFIG_BT_CENTRAL is defined, then connected and disconnected HCI events will be included and there will be calls to it. But the condition will never be true if not CONFIG_BT_PERIPHERAL is also enabled. So I think carles' suggestion is correct.

Reply:
Given the same argument why this must be under BT_PERIPHERAL, then this will only be called for connectable advertising. So changed to assert on this assumption, and removed all if(conn) in this function.